yt_dlp/extractor/youtube/pot/builtin/cache/memory.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import datetime as dt
+import typing
+from collections import OrderedDict
+from threading import Lock
+
+from yt_dlp.extractor.youtube.pot._provider import BuiltInIEContentProvider
+from yt_dlp.extractor.youtube.pot.cache import (
+    PoTokenCacheProvider,
+    register_pcp,
+    register_pcp_preference,
+)
+from yt_dlp.globals import _pot_memory_cache
+from yt_dlp.utils import int_or_none
+
+
+def initialize_global_cache(max_size: int):
+    if _pot_memory_cache.value.get('cache') is None:
+        _pot_memory_cache.value['cache'] = OrderedDict()
+        _pot_memory_cache.value['lock'] = Lock()
+        _pot_memory_cache.value['max_size'] = max_size
+
+    if _pot_memory_cache.value['max_size'] != max_size:
+        raise ValueError('Cannot change max_size of initialized global memory cache')
+
+    return _pot_memory_cache.value['cache'], _pot_memory_cache.value['lock'], _pot_memory_cache.value['max_size']
+
+
+@register_pcp
+class MemoryLRUPCP(PoTokenCacheProvider, BuiltInIEContentProvider):
```

Comment:
1. Why do we not want to cache to disk?
2. Why not use python's `lru_cache` with us only doing the expires check

Reply:
> our current disk cache is not optimal/sufficient for PO Tokens atm. 

Can you explain? My thinking is to just write the cache to disk (as one file) on close and load on init. The LRU rotations can happen just in memory.

Reply:
Yes that would work but you would need locking which our cache does not have currently (this needs to work with concurrent yt-dlp instances)

I had tested a similar thing in a plugin (no locking) https://github.com/coletdjnz/yt-dlp-getpot-wpc/blob/main/yt_dlp_plugins/extractor/getpot_wpc.py#L135

Anyways, this is something to look into a followup PR, I've considered it out of scope for this one due to the complexity involved (see the PR description)